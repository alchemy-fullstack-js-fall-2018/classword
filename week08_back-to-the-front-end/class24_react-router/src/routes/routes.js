import React from 'react';
import Search from '../components/search/Search';
import ArtistDetail from '../components/artist/ArtistDetail';

export const ROUTES = {
  SEARCH: {
    path: '/search',
    Component: Search,
    linkTo: () => '/search'
  },
  HI: {
    path: '/hi',
    Component: () => <h1>hi</h1>,
    linkTo: () => '/hi'
  },
  ARTIST: {
    path: '/artists/:id',
    Component: ArtistDetail,
    linkTo: id => `/artists/${id}`
  }
};
