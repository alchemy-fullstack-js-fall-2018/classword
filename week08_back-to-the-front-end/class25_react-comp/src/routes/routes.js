import EpisodeList from '../components/episodes/EpisodeList';

export const ROUTES = {
  EPISODES: {
    path: '/episodes',
    Component: EpisodeList,
    linkPath: () => '/episodes'
  },
  EPISODE: {
    path: '/episodes/:id',
    Component: () => { }, // TODO: Write component
    linkPath: id => `/episodes/${id}`
  }
};
