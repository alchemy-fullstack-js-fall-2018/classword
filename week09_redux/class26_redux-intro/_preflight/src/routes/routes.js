import Home from '../components/home/Home';
import NoteContainer from '../containers/notes/NoteContainer';
import HiNotes from '../containers/notes/HiNotes';
import NotesByLength from '../containers/notes/NotesByLength';
import NotesByTitle from '../containers/notes/NotesByTitle';

export const ROUTES = {
  HOME: {
    path: '/',
    Component: Home,
    linkTo: () => '/'
  },
  NOTE: {
    path: '/note/:id',
    Component: NoteContainer,
    linkTo: id => `/note/${id}`
  },
  NOTES_BY_TITLE: {
    path: '/notes/byTitle',
    Component: NotesByTitle,
    linkTo: () => '/notes/byTitle'
  },
  NOTES_BY_LENGTH: {
    path: '/notes/byLength',
    Component: NotesByLength,
    linkTo: () => '/notes/byLength'
  },
  HI_NOTES: {
    path: '/notes/hi',
    Component: HiNotes,
    linkTo: () => '/notes/hi'
  }
};
