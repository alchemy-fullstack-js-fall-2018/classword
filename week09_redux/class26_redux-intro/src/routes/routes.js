import Home from '../components/home/Home';
import NoteContainer from '../containers/notes/NoteContainer';

export const ROUTES = {
  HOME: {
    path: '/',
    Component: Home,
    linkTo: () => '/'
  },
  NOTE: {
    path: '/notes/:id',
    Component: NoteContainer,
    linkTo: id => `/notes/${id}`
  }
};
