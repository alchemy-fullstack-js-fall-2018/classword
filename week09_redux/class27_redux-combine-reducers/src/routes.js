import AllPizzas from './containers/pizza/AllPizzas';
import AllTransactions from './containers/transactions/AllTransactions';
export default {
  ALL_PIZZA: {
    path: '/',
    Component: AllPizzas,
    linkTo: () => '/'
  },
  TRANSACTIONS: {
    path: '/transactions',
    Component: AllTransactions,
    linkTo: () => '/transactions'
  }
};
