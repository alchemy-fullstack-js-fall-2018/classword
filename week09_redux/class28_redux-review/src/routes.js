import FilteredClothes from './containers/clothes/FilteredClothes';

export default {
  CLOTHES: {
    path: '/clothes',
    Component: FilteredClothes,
    linkTo: () => '/clothes'
  }
};
