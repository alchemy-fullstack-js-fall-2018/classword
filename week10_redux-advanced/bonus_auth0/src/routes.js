import { withAuth } from './auth/withAuth';
import Home from './components/home/Home';
import ArtistsSearch from './containers/artists/ArtistsSearch';
import { Login, Callback, Signup } from './components/auth/Auth';

export const ROUTES = {
  HOME: {
    path: '/',
    Component: Home,
    linkTo: () => '/'
  },
  SIGNUP: {
    path: '/signup',
    Component: Signup,
    linkTo: () => '/signup'
  },
  LOGIN: {
    path: '/login',
    Component: Login,
    linkTo: () => '/login'
  },
  CALLBACK: {
    path: '/callback',
    Component: Callback,
    linkTo: () => '/callback'
  },
  ARTISTS: {
    path: '/artists',
    Component: withAuth(ArtistsSearch),
    linkTo: () => '/artists'
  }
};
