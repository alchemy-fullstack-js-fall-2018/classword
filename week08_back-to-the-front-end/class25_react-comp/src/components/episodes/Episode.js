import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';

export default function Episode({ id, title, episodeNumber }) {
  return (
    <div>
      <h3>{title}</h3>
      <h4>{episodeNumber}</h4>
      <Link to={ROUTES.EPISODE.linkPath(id)}>Link</Link>
    </div>
  );
}
