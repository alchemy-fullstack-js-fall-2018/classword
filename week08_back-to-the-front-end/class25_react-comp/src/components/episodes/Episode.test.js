import React from 'react';
import { shallow } from 'enzyme';
import Episode from './Episode';
import { MemoryRouter } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';

describe('Episode', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Episode id="1234" title="My Episode" />);
  });

  it('renders an Episode', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays the title', () => {
    const wrapperWithRouter = shallow(
      <MemoryRouter>
        <Episode id="1234" title="My Episode" />
      </MemoryRouter>
    );
    expect(wrapperWithRouter.html()).toContain('My Episode');
  });

  it('links to an EpisodeDetail', () => {
    expect(wrapper.find('Link').props().to).toEqual(ROUTES.EPISODE.linkPath('1234'));
  });
});
