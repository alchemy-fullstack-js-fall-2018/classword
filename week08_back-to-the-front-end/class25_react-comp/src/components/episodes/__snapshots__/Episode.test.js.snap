// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode renders an Episode 1`] = `
<div>
  <h3>
    My Episode
  </h3>
  <h4 />
  <Link
    replace={false}
    to="/episodes/1234"
  >
    Link
  </Link>
</div>
`;
